### PR TITLE
pullrequest-init: Add getters to make PullRequest nil-safe.

### PR DESCRIPTION
**Pull request by wlynch**
# Changes

Adds getters to all funcs so that libraries can safely handle the struct
without needing to worry about doing nil checks to avoid panics. This is
heavily influenced by the behavior of Go protobufs and other libraries
like https://github.com/google/go-github.

Also adds test coverage for uploadComments and uploadLabels for nil
lists.

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/pipeline/blob/master/CONTRIBUTING.md)
for more details._

# Release Notes

None.

### CODE REVIEW

File: cmd/pullrequest-init/fake_github_test.go [hunk 1]
```diff
@@ -41,7 +41,10 @@ func TestFakeGitHubComments(t *testing.T) {
 		t.Fatalf("List Issues: wanted [], got %+v, %+v, %v", got, resp, err)
 	}
 
-	if _, _, err := client.Issues.CreateComment(ctx, owner, repo, prNum, comment); err != nil {
+	comment, _, err := client.Issues.CreateComment(ctx, owner, repo, prNum, &github.IssueComment{
```
Comment (bobcatfish) [inline cmd/pullrequest-init/fake_github_test.go]: i might just not be seeing it, but I can't find where the `comment` var was declared before this change :O Was this a bug in the tests? If so that would mean they aren't running in CI and there's no linting happening... 😨

Comment (vdemeester) [reply cmd/pullrequest-init/fake_github_test.go]: 😨 😨 😨 😨 😨 😨 😨 😨

File: cmd/pullrequest-init/github.go [hunk 2]
```diff
@@ -287,13 +292,22 @@ func (h *GitHubHandler) createNewComments(ctx context.Context, comments []*Comme
...
+		h.Logger.Infof("Creating comment %s for PR %d", dc.GetText(), h.prNum)
 		if _, _, err := h.Issues.CreateComment(ctx, h.owner, h.repo, h.prNum, c); err != nil {
 			h.Logger.Warnf("Error creating comment: %v", err)
 			merr = multierror.Append(merr, err)
 		}
 	}
 	return merr
 }
+
+func githubCommentToTekton(c *github.IssueComment, pathPrefix string) *Comment {
```
[the remark this answers is not in the document]
Comment (vdemeester) [reply cmd/pullrequest-init/github.go]: right, if it's only used for the test, let's add it in test files.

**Comment by bobcatfish** [comment]
We ended up running into an issue in our 0.5.0 release (#1075 ) that was due to trying to access an attribute on a nil value - I agree with what @vdemeester said about using getters not being very golang-y (is there a proper word for that?) but I also wonder if we can make our types a bit more safe.

Seems like this isn't the direction we want to go at the moment tho 🤔

**Comment by bobcatfish** [comment]
I'm going to close this for now assuming we don't want to go in this direction but please re-open if I'm wrong!
